File: ClientSide/app/scripts/service/UserService.js
/**
 * Created by alex_prozorov on 20.09.2014.
 */
(function () {
    'use strict';

    var UserService = function (Restangular) {
        this.Restangular = Restangular;
    };

    UserService.$inject = ['Restangular'];

    UserService.prototype = {

        getUserId: function (){
            var u = this.Restangular.all('user').getList();
            return u;
        }
    };

    angular.module('auction').service('UserService', UserService);
}());

File: ServerSide/auction_jaxrs/src/main/java/com/farata/course/mwd/auction/service/UserService.java
package com.farata.course.mwd.auction.service;

import com.farata.course.mwd.auction.data.DataEngine;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;

/**
 * Created by prozorov on 18/09/14.
 */

@Path("user")
@Produces("application/json")
public class UserService {

    DataEngine dataEngine;

    @Inject
    public void setDataEngine(DataEngine dataEngine) {
        this.dataEngine = dataEngine;
    }

    @GET
    public String getMyUserId() {
        return "[{\"userId\": " + dataEngine.getUserRepository().registerUser() + "}]";
    }
}
